// sendblue/dist/base/Message.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MessageImpl = void 0;
var MessageImpl = /** @class */ (function () {
    function MessageImpl(message) {
        this.accountEmail = message.accountEmail;
        this.content = message.content;
        this.is_outbound = message.is_outbound;
        this.status = message.status;
        this.error_code = message.error_code;
        this.error_message = message.error_message;
        this.message_handle = message.message_handle;
        this.date_sent = message.date_sent;
        this.date_updated = message.date_updated;
        this.from_number = message.from_number;
        this.number = message.number;
        this.to_number = message.to_number;
        this.was_downgraded = message.was_downgraded;
        this.plan = message.plan;
        this.media_url = message.media_url;
        this.message_type = message.message_type;
        this.group_id = message.group_id;
        this.participants = message.participants;
    }
    MessageImpl.from = function (data) {
        return new MessageImpl(data);
    };
    return MessageImpl;
}());
exports.MessageImpl = MessageImpl;
