// sendblue/dist/base/SendblueClient.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SendblueClient = void 0;
var group_message_1 = require("../rest/group-message");
var message_1 = require("../rest/message");
var RequestClient_1 = require("./RequestClient");
var SendblueClient = /** @class */ (function () {
    function SendblueClient(apiKey, apiSecret, options) {
        if (options === void 0) { options = {}; }
        var _a;
        this.logLevel = 'info';
        if (options.logLevel !== ((_a = options === null || options === void 0 ? void 0 : options.axiosClientOptions) === null || _a === void 0 ? void 0 : _a.logLevel)) {
            if (!options.axiosClientOptions)
                options.axiosClientOptions = {};
            options.axiosClientOptions.logLevel = options.logLevel;
        }
        this.logLevel = options.logLevel || 'info';
        this.axiosClient = new RequestClient_1.RequestClient((options === null || options === void 0 ? void 0 : options.axiosClientOptions) || {}, apiKey, apiSecret);
        this.groups = new group_message_1.Groups(this);
        this.messages = new message_1.Messages(this);
    }
    SendblueClient.prototype.request = function (options) {
        return this.axiosClient.request(options);
    };
    SendblueClient.prototype.sendGroupMessage = function (options) {
        return this.groups.sendGroupMessage(options);
    };
    SendblueClient.prototype.sendMessage = function (options) {
        return this.messages.sendMessage(options);
    };
    return SendblueClient;
}());
exports.SendblueClient = SendblueClient;
